snuba/manual_jobs/runner.py

```diff
@@ -34,7 +63,69 @@ def _build_job_spec_from_entry(content: Any) -> JobSpec:
     return job_spec
 
 
+def _job_lock_key(job_id: str) -> str:
+    return f"snuba:manual_jobs:{job_id}:lock"
+
+
+def _job_status_key(job_id: str) -> str:
+    return f"snuba:manual_jobs:{job_id}:execution_status"
+
+
+def _acquire_job_lock(job_id: str) -> bool:
+    return bool(
+        _redis_client.set(
+            name=_job_lock_key(job_id), value=1, nx=True, ex=(24 * 60 * 60)
+        )
+    )
+
+
+def _release_job_lock(job_id: str) -> None:
+    _redis_client.delete(_job_lock_key(job_id))
+
+
+def _set_job_status(job_id: str, status: JobStatus) -> JobStatus:
+    _redis_client.set(name=_job_status_key(job_id), value=status.value)
+    return status
```

Comment:
it's just to make the chaining on lines 114, 120, and 126 eaier. e.g.  instead of

```
current_job_status = JobStatus.FINISHED
_set_job_status(job_spec.job_id, JobStatus.FINISHED)
```

you have

```
current_job_status = _set_job_status(job_spec.job_id, JobStatus.FINISHED)
```

if you prefer the verbose way, I can change it

Reply:
I don't think there's much value to returning the result of set(). If anything, I'd probably throw an exception if set fails

Reply:
Yes, the failure case made me question the return value. Probably better to throw the exception if `set()` fails, otherwise return the `status`.